tests/unit/test_iam_dsql_plugin.py

```diff
@@ -0,0 +1,470 @@
+    else:
+        initial_token = TokenInfo(_TEST_TOKEN, datetime.now() - timedelta(minutes=5))
+    cache_key: str = IamAuthUtils.get_cache_key(
+            user,
+            host,
+            port,
+            region
+        )
+    _token_cache[cache_key] = initial_token
+
+    return cache_key, initial_token
+
+
+@pytest.mark.parametrize("user", [
+    pytest.param("admin"),
+    pytest.param("non-admin"),
+])
+@patch("aws_advanced_python_wrapper.iam_plugin.IamAuthPlugin._token_cache", _token_cache)
+def test_pg_connect_valid_token_in_cache(user, mocker, mock_plugin_service, mock_session, mock_func, mock_client, mock_dialect):
+    test_props: Properties = Properties({"user": user})
+    cache_key, _ = set_token_cache(user, _PG_HOST_INFO.url, _DEFAULT_PG_PORT, _PG_REGION)
+
+    target_plugin: IamAuthPlugin = IamAuthPlugin(mock_plugin_service,
+                                                 DSQLTokenUtils(),
+                                                 mock_session)
+    target_plugin.connect(
+        target_driver_func=mocker.MagicMock(),
+        driver_dialect=mock_dialect,
+        host_info=_PG_HOST_INFO,
+        props=test_props,
+        is_initial_connection=False,
+        connect_func=mock_func)
+
+    actual_token = _token_cache.get(cache_key)
+    if user == "admin":
+        mock_client.generate_db_connect_admin_auth_token.assert_not_called()
+        assert _GENERATED_TOKEN != actual_token.token
+    else:
+        mock_client.generate_db_connect_auth_token.assert_not_called()
+        assert _GENERATED_TOKEN_NON_ADMIN != actual_token.token
+
+    assert _TEST_TOKEN == actual_token.token
+    assert actual_token.is_expired() is False
+
+
+@patch("aws_advanced_python_wrapper.iam_plugin.IamAuthPlugin._token_cache", _token_cache)
+def test_pg_connect_with_invalid_port_fall_backs_to_host_port(
+        mocker, mock_plugin_service, mock_session, mock_func, mock_client, mock_dialect):
+    test_props: Properties = Properties({"user": "admin"})
+    invalid_port = "0"
+    test_props[WrapperProperties.IAM_DEFAULT_PORT.name] = invalid_port
+
+    # Assert no password has been set
+    assert test_props.get("password") is None
+
+    target_plugin: IamAuthPlugin = IamAuthPlugin(mock_plugin_service,
+                                                 DSQLTokenUtils(),
+                                                 mock_session)
+    target_plugin.connect(
+        target_driver_func=mocker.MagicMock(),
+        driver_dialect=mock_dialect,
+        host_info=_PG_HOST_INFO_WITH_PORT,
+        props=test_props,
+        is_initial_connection=False,
+        connect_func=mock_func)
+
+    mock_client.generate_db_connect_admin_auth_token.assert_called_with(
+        _PG_HOST_INFO.url, _PG_REGION
+    )
+
+    actual_token = _token_cache.get(f"{_PG_REGION}:{_PG_HOST_INFO.url}:1234:admin")
+    assert _GENERATED_TOKEN == actual_token.token
+    assert actual_token.is_expired() is False
```

Comment:
nit, same suggestion for the other lines with this pattern in the file:
```suggestion
    assert not actual_token.is_expired()
```

Reply:
Updated all the occurrences.